src/expr/src/scalar/func/macros.rs

```diff
@@ -106,7 +106,7 @@ macro_rules! sqlfunc {
                             .try_into()
                             .expect("expression already typechecked");
 
-                        super::$fn_name(a).map(|r| r.into())
+                        super::$fn_name(a).map(move |r| WithArena::new(temp_storage, r).into())
```

Comment:
As this macro gets more complex, do you think it's possible to keep inline docs around what occurring here? Took me a few min to page some of this back in and am a little worried that the bus factor here is pretty high.